### PR TITLE
[INLONG-10644][Sort] Add the elasticsearch-base module to implement elasticsearch connector 6&7

### LINKED ISSUE #10644: [Feature][Sort] The base common of elasticsearch connector is neccessary.
### Description

Before implement the elasticsearch connector,  some base common classes and interface have to implement.

### Use case

The elasticsearch connector 6/7 can be implement easily through this base module.

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by XiaoYou201**
### [INLONG-10644][Sort] Add the elasticsearch-base module to implement elasticsearch connector 6&7

Fixes #10644 

### Motivation

Add the elasticsearch-base module to implement elasticsearch connector 6&7

To easy to develop

### Modifications

Add some public constants, methods, classes, interfaces, etc.

### CODE REVIEW

**Comment by EMsnap** [comment]
Please add test result on es 6 or 7

File: licenses/inlong-sort-connectors/LICENSE [hunk 1]
```diff
@@ -935,6 +935,28 @@ License : https://github.com/ververica/flink-cdc-connectors/blob/master/LICENSE
...
+1.3.29 inlong-sort/sort-flink/sort-flink-v1.18/sort-connectors/elasticsearch-base/src/main/java/org/apache/inlong/sort/elasticsearch/ActionRequestFailureHandler.java
+1.3.29 inlong-sort/sort-flink/sort-flink-v1.18/sort-connectors/elasticsearch-base/src/main/java/org/apache/inlong/sort/elasticsearch/BufferingNoOpRequestIndexer.java
+1.3.29 inlong-sort/sort-flink/sort-flink-v1.18/sort-connectors/elasticsearch-base/src/main/java/org/apache/inlong/sort/elasticsearch/ElasticsearchApiCallBridge.java
+1.3.29 inlong-sort/sort-flink/sort-flink-v1.18/sort-connectors/elasticsearch-base/src/main/java/org/apache/inlong/sort/elasticsearch/ElasticsearchSinkBase.java
+1.3.29 inlong-sort/sort-flink/sort-flink-v1.18/sort-connectors/elasticsearch-base/src/main/java/org/apache/inlong/sort/elasticsearch/ElasticsearchSinkFunction.java
+1.3.29 inlong-sort/sort-flink/sort-flink-v1.18/sort-connectors/elasticsearch-base/src/main/java/org/apache/inlong/sort/elasticsearch/RequestIndexer.java
+Source  : org.apache.flink:flink-connector-jdbc-1.15.4.jar (Please note that the software have been modified.)
```
Comment (aloyszhang) [inline licenses/inlong-sort-connectors/LICENSE:957]: plz correct the source information

**Comment by aloyszhang** [comment]
Plz add source information for es connector classes

**Comment by XiaoYou201** [comment]
> Plz add source information for es connector classes

The es connector 6/7 are base on it. So,in my opinion, it is also a part of es connector.

File: licenses/inlong-sort-connectors/LICENSE (hunk 1 above)
Comment (XiaoYou201) [reply licenses/inlong-sort-connectors/LICENSE]: fix it

**Comment by XiaoYou201** [comment]
> Please add test result on es 6 or 7
As follow image, I implement es 6 connector and  postgreSQL to ElasticSearch6 test successfully . But es6 base on this pr, so, the es6 connector pr will be submitted after  this pr merged.
image
image
image

File: licenses/inlong-sort-connectors/LICENSE [hunk 2]
```diff
@@ -938,6 +938,28 @@ License : https://github.com/ververica/flink-cdc-connectors/blob/master/LICENSE
...
+1.3.29 inlong-sort/sort-flink/sort-flink-v1.18/sort-connectors/elasticsearch-base/src/main/java/org/apache/inlong/sort/elasticsearch/util/RetryRejectedExecutionFailureHandler.java
+1.3.29 inlong-sort/sort-flink/sort-flink-v1.18/sort-connectors/elasticsearch-base/src/main/java/org/apache/inlong/sort/elasticsearch/ActionRequestFailureHandler.java
+1.3.29 inlong-sort/sort-flink/sort-flink-v1.18/sort-connectors/elasticsearch-base/src/main/java/org/apache/inlong/sort/elasticsearch/BufferingNoOpRequestIndexer.java
+1.3.29 inlong-sort/sort-flink/sort-flink-v1.18/sort-connectors/elasticsearch-base/src/main/java/org/apache/inlong/sort/elasticsearch/ElasticsearchApiCallBridge.java
+1.3.29 inlong-sort/sort-flink/sort-flink-v1.18/sort-connectors/elasticsearch-base/src/main/java/org/apache/inlong/sort/elasticsearch/ElasticsearchSinkBase.java
+1.3.29 inlong-sort/sort-flink/sort-flink-v1.18/sort-connectors/elasticsearch-base/src/main/java/org/apache/inlong/sort/elasticsearch/ElasticsearchSinkFunction.java
+1.3.29 inlong-sort/sort-flink/sort-flink-v1.18/sort-connectors/elasticsearch-base/src/main/java/org/apache/inlong/sort/elasticsearch/RequestIndexer.java
```
Comment (aloyszhang) [inline licenses/inlong-sort-connectors/LICENSE:959]:
```suggestion
       inlong-sort/sort-flink/sort-flink-v1.18/sort-connectors/elasticsearch-base/src/main/java/org/apache/inlong/sort/elasticsearch/table/ElasticsearchConfiguration.java
       inlong-sort/sort-flink/sort-flink-v1.18/sort-connectors/elasticsearch-base/src/main/java/org/apache/inlong/sort/elasticsearch/table/ElasticsearchConnectorOptions.java
       inlong-sort/sort-flink/sort-flink-v1.18/sort-connectors/elasticsearch-base/src/main/java/org/apache/inlong/sort/elasticsearch/table/ElasticsearchValidationUtils.java
       inlong-sort/sort-flink/sort-flink-v1.18/sort-connectors/elasticsearch-base/src/main/java/org/apache/inlong/sort/elasticsearch/table/IndexGenerator.java
       inlong-sort/sort-flink/sort-flink-v1.18/sort-connectors/elasticsearch-base/src/main/java/org/apache/inlong/sort/elasticsearch/table/IndexGeneratorBase.java
       inlong-sort/sort-flink/sort-flink-v1.18/sort-connectors/elasticsearch-base/src/main/java/org/apache/inlong/sort/elasticsearch/table/IndexGeneratorFactory.java
       inlong-sort/sort-flink/sort-flink-v1.18/sort-connectors/elasticsearch-base/src/main/java/org/apache/inlong/sort/elasticsearch/table/KeyExtractor.java
       inlong-sort/sort-flink/sort-flink-v1.18/sort-connectors/elasticsearch-base/src/main/java/org/apache/inlong/sort/elasticsearch/table/RowElasticsearchSinkFunction.java
       inlong-sort/sort-flink/sort-flink-v1.18/sort-connectors/elasticsearch-base/src/main/java/org/apache/inlong/sort/elasticsearch/table/StaticIndexGenerator.java
       inlong-sort/sort-flink/sort-flink-v1.18/sort-connectors/elasticsearch-base/src/main/java/org/apache/inlong/sort/elasticsearch/util/IgnoringFailureHandler.java
       inlong-sort/sort-flink/sort-flink-v1.18/sort-connectors/elasticsearch-base/src/main/java/org/apache/inlong/sort/elasticsearch/util/NoOpFailureHandler.java
       inlong-sort/sort-flink/sort-flink-v1.18/sort-connectors/elasticsearch-base/src/main/java/org/apache/inlong/sort/elasticsearch/util/RetryRejectedExecutionFailureHandler.java
       inlong-sort/sort-flink/sort-flink-v1.18/sort-connectors/elasticsearch-base/src/main/java/org/apache/inlong/sort/elasticsearch/ActionRequestFailureHandler.java
       inlong-sort/sort-flink/sort-flink-v1.18/sort-connectors/elasticsearch-base/src/main/java/org/apache/inlong/sort/elasticsearch/BufferingNoOpRequestIndexer.java
       inlong-sort/sort-flink/sort-flink-v1.18/sort-connectors/elasticsearch-base/src/main/java/org/apache/inlong/sort/elasticsearch/ElasticsearchApiCallBridge.java
       inlong-sort/sort-flink/sort-flink-v1.18/sort-connectors/elasticsearch-base/src/main/java/org/apache/inlong/sort/elasticsearch/ElasticsearchSinkBase.java
       inlong-sort/sort-flink/sort-flink-v1.18/sort-connectors/elasticsearch-base/src/main/java/org/apache/inlong/sort/elasticsearch/ElasticsearchSinkFunction.java
       inlong-sort/sort-flink/sort-flink-v1.18/sort-connectors/elasticsearch-base/src/main/java/org/apache/inlong/sort/elasticsearch/RequestIndexer.java
```

**Comment by aloyszhang** [comment]
Flink-1.15 needs this feature too, would you like to work on it?

**Comment by XiaoYou201** [comment]
> BTW, Flink-1.15 needs this feature too, would you like to work on it?

yes, I would like to do that, it will done soon~

**Review by aloyszhang — APPROVED** [review]

**Review by vernedeng — APPROVED** [review]

**Review by EMsnap — APPROVED** [review]

**Pull request merged**